components/match2/wikis/zula/match_group_input_custom.lua

```diff
@@ -0,0 +1,683 @@
+	local opponents = {}
+	local isScoreSet = false
+	for opponentIndex = 1, MAX_NUM_OPPONENTS do
+		-- read opponent
+		local opponent = match['opponent' .. opponentIndex]
+		if not Logic.isEmpty(opponent) then
+			CustomMatchGroupInput.processOpponent(opponent, match.date)
+
+			-- Retrieve icon for team
+			if opponent.type == Opponent.team then
+				opponent.icon, opponent.icondark = opponentFunctions.getIcon(opponent.template)
+			end
+
+			-- apply status
+			if TypeUtil.isNumeric(opponent.score) then
+				opponent.status = 'S'
+				isScoreSet = true
+			elseif Table.includes(ALLOWED_STATUSES, opponent.score) then
+				opponent.status = opponent.score
+				opponent.score = -1
+			end
+			opponents[opponentIndex] = opponent
+
+			-- get players from vars for teams
+			if opponent.type == Opponent.team and not Logic.isEmpty(opponent.name) then
+				match = matchFunctions.getPlayers(match, opponentIndex, opponent.name)
+			end
+		end
+	end
+
+	-- Handle tournament status for unfinished matches
+	if (not Logic.readBool(match.finished)) and Logic.isNotEmpty(match.status) then
+		match.finished = match.status
+	end
+
+	if Table.includes(NP_MATCH_STATUS, match.finished) then
+		match.resulttype = 'np'
+		match.status = match.finished
+		match.finished = false
+		match.dateexact = false
+	else
+		-- see if match should actually be finished if score is set
+		if isScoreSet and not Logic.readBool(match.finished) and match.hasDate then
+			local currentUnixTime = os.time(os.date('!*t'))
+			local lang = mw.getContentLanguage()
+			local matchUnixTime = tonumber(lang:formatDate('U', match.date))
+			local threshold = match.dateexact and 30800 or 86400
+			if matchUnixTime + threshold < currentUnixTime then
+				match.finished = true
+			end
+		end
+
+		if Logic.readBool(match.finished) then
+			match, opponents = CustomMatchGroupInput.getResultTypeAndWinner(match, opponents)
+		end
+	end
+
+	-- Update all opponents with new values
+	for opponentIndex, opponent in pairs(opponents) do
+		match['opponent' .. opponentIndex] = opponent
+	end
+	return match
+end
+
+-- Get Playerdata from Vars (get's set in TeamCards)
+function matchFunctions.getPlayers(match, opponentIndex, teamName)
+	-- match._storePlayers will break after the first empty player. let's make sure we don't leave any gaps.
+	local count = 1
+	for playerIndex = 1, MAX_NUM_PLAYERS do
+		-- parse player
+		local player = match['opponent' .. opponentIndex .. '_p' .. playerIndex] or {}
+		player = Json.parseIfString(player)
+		local playerPrefix = teamName .. '_p' .. playerIndex
+		player.name = player.name or Variables.varDefault(playerPrefix)
+		player.flag = player.flag or Variables.varDefault(playerPrefix .. 'flag')
+		player.displayname = player.displayname or Variables.varDefault(playerPrefix .. 'dn')
+		if not Table.isEmpty(player) then
+			match['opponent' .. opponentIndex .. '_p' .. count] = player
+			count = count + 1
+		end
+	end
+	return match
+end
```

Comment:
use comons?

Reply:
were both commons meant I can just remove those lines? (the newest revision had both open comments removed and there were no errors visually)